### PR TITLE
Cancellations

### PR DESCRIPTION
**Pull request by Dkendal**
Adds support for cancellations on payments. Internally it calls Adyen's cancel_or_refund end point so it doesn't matter what the original state of the payment was.

### CODE REVIEW

**Comment by Dkendal** [comment]
jenkins build this please

**Comment by Dkendal** [comment]
jenkins please

**Comment by Dkendal** [comment]
jenkins build this please

**Comment by Dkendal** [comment]
test this please

**Comment by Dkendal** [comment]
jenkinsssssssss

**Comment by Dkendal** [comment]
test this please

**Comment by Dkendal** [comment]
jenkins build this please

**Comment by Dkendal** [comment]
test this please

File: app/models/adyen_notification.rb [hunk 1]
```diff
@@ -84,5 +101,43 @@ def actions
...
+      REFUNDED_REVERSED
+    ].member? self.event_code
+  end
+
+  def normal_event?
+    AUTHORISATION == self.event_code
+  end
+
+  def bank_transfer?
+    self.payment_method.match(/^bankTransfer/)
```
Comment (bsodmike) [inline app/models/adyen_notification.rb:121]: You could eliminate the need for a match object and do `self.payment_method =~ /^bankTransfer/`; no match will return nil.

File: app/models/adyen_notification.rb [hunk 2]
```diff
@@ -84,5 +101,43 @@ def actions
...
+      psp_reference: self.psp_reference,
+      event_code: self.event_code,
+      success: self.success
+    )
+  end
+
+  def auto_captured?
+    # todo include bank transfers
+    payment_method_auto_capture_only? ||
+      bank_transfer?
```
Comment (bsodmike) [inline app/models/adyen_notification.rb]: Should be ok in a single line?

File: app/models/adyen_notification.rb (hunk 1 above)
Comment (Dkendal) [inline app/models/adyen_notification.rb:121]: I don't really see a semantic difference.  `=~` is just syntax sugar on `str.match(regex).begin(0)`.

File: app/models/adyen_notification.rb (hunk 2 above)
Comment (Dkendal) [inline app/models/adyen_notification.rb]: :+1:

**Pull request merged**